test/dsa_no_digest_size_test.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "testutil.h"
+
+#include <openssl/evp.h>
+#include <openssl/dsa.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+static DSA *dsa;
+
+static int genkeys(void) {
+    return TEST_ptr(dsa = DSA_new())
+        && TEST_int_eq(DSA_generate_parameters_ex(
```

Comment:
Please write this in the cascading if-fail style, as used elsewhere.
Put the DSA_generate call on a separate line and you can probably fix the indent/linebreak stuff.